texar/utils/utils.py

```diff
@@ -605,6 +608,37 @@ def get_instance_kwargs(kwargs: Kwargs, hparams: ParamDict) -> Kwargs:
     return kwargs_
 
 
+def get_output_size(instance_: Optional[Module]) -> Optional[int]:
+    r"""Return the final dimension of output size from :attr:`instance_`.
+
+    If type of :attr:`instance_` is among the common types, the final
+    dimension of output size will be computed.
+
+    Args:
+        instance_: A :class:`~torch.nn.Module` instance from
+            which to compute output size.
+
+    :returns:
+        int (optional): The final dimension of the output size.
+    """
+    types = {
+        nn.Linear: lambda x: x.out_features,
+        nn.Bilinear: lambda x: x.out_features,
+        nn.Conv1d: lambda x: x.out_channels * len(x.kernel_size),
+        nn.Conv2d: lambda x: x.out_channels * len(x.kernel_size),
+        nn.Conv3d: lambda x: x.out_channels * len(x.kernel_size),
+        nn.Embedding: lambda x: x.embedding_dim,
+        nn.EmbeddingBag: lambda x: x.embedding_dim,
+        nn.RNNCell: lambda x: x.hidden_size,
+        nn.LSTMCell: lambda x: x.hidden_size,
+        nn.GRUCell: lambda x: x.hidden_size,
+    }
+    for t, l in types.items():
+        if isinstance(instance_, t):
```

Comment:
Consider replacing `Conv*d` with the more general `_ConvNd` base class. Since `_ConvNd` is also the base class for `ConvTranspose*d` classes, please also take a look at their code to make sure the output size calculation is correct. If their sizes are not computed in the same way, let's keep this part as is.

The `*Cell` classes can be with `RNNCellBase` without errors.